components/net/http_loader.rs

```diff
@@ -1202,7 +1201,14 @@ fn http_network_fetch(
     // This will only get the headers, the body is read later
     let (res, msg) = match response_future.wait() {
         Ok(wrapped_response) => wrapped_response,
-        Err(error) => return Response::network_error(error),
+        Err(error) => {
+            context
+                .timing
+                .lock()
+                .unwrap()
+                .set_attribute(ResourceAttribute::ResponseEnd);
+            return Response::network_error(error);
```

Comment:
Rather than adding this code to each place where we could return, I think we should create a `struct ResponseEndTimer(Option<Arc<Mutex<ResourceFetchTiming>>>)` that has a `Drop` implementation which calls `set_attribute`. This will ensure that whenever the value goes out of scope, we set the timing information appropriately. We can also add a `neuter` method which sets the stored value to None, and call that when the connection is made successfully. Does that make sense?

Reply:
I think so! Being new to `Rust`, this pattern is not familiar to me, but it makes sense (and sounds pretty cool). I'll give it a go later tonight or tomorrow, probably.